### DISCUSSION TITLE
SMTP Setup with MailJet (No Errors... but no emails)

### DISCUSSION THREAD

**Discussion by jscheller** [pocketbase/pocketbase#3795] [Q&A] [answered]
Hello!

Working with PocketBase 0.18.9 in a dev environment inside a docker container and trying to get the standard emails out through MailJet’s SMTP service. I can configure SMTP via the Mail Settings in the admin interface, and when I click “Send test email” after a short delay, get the little green “Successfully sent test email” popup, so no errors. But no emails come through.

Tried the easy stuff… different destination addresses, checked spam folders, changed subject line of mail template, etc.

I can monitor via MailJet’s web interface to see when emails are pushed through their SMTP service, so I created a python script inside the docker container to send via SMTP using all the same settings and it sends/delivers just fine. I can confirm in MailJet’s web UI that python script pushing emails out from the container runs just fine, and indeed they get delivered. 

However, I get nothing at MailJet’s web UI (or ultimately in my inbox), and no errors anywhere when I send from PocketBase. Messages sent through PocketBase do not show in MailJet’s interface reporting SMTP events. So I’m a bit lost on how to debug.

I’ve tried different variants of ports and login methods in PocketBase, all to no effect. If I tweak the user/password for the mail server, I do get an auth error so I’m pretty sure PocketBase is making a connection to MailJet’s SMTP servers just fine. Feels like I’m just missing something dumb here.

My testing python script that gets messages out is below… Any help or insight is appreciated!

Many Thanks,

Jim

<img width="895" alt="Screenshot 2023-11-20 at 1 35 51 PM" src="https://github.com/pocketbase/pocketbase/assets/1310028/ba0f0dc9-8213-40aa-99ca-5742c27bde5b">

```
import smtplib
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText

fromAddr = 'noreply@wordsearchwizard.com'
toAddr = xxxxxxxxx@jamesscheller.com'  #redacted dest email  

msg = MIMEMultipart()
msg['From'] = fromAddr
msg['To'] = toAddr
msg['Subject'] = 'another simple email in python'
message = 'here is the email'
msg.attach(MIMEText(message))

with smtplib.SMTP('in-v3.mailjet.com',587) as mail_server:
    mail_server.ehlo()
    mail_server.starttls()
    mail_server.ehlo()
    mail_server.login('6324a8xxxxxx', '7d2f01xxxxxxxx')  #redacted user/pw  
    mail_server.sendmail(fromAddr,toAddr,msg.as_string())

```

**Comment by ganigeorgiev** [comment]
Are you sure that you use the same source and dest emails when testing with PocketBase?

Is the email you use in PocketBase as "Sender address" (_available at the top of the mail settings page_) validated/registered in Mailjet?

The success message usually means that PocketBase has successfully submitted the message to the mail server but whether it will be actually delivered to the recipient would depend on the smtp host or the recipient server filters.

If you are still not able to resolve it, could you paste a redacted version of the raw mail sent with the python script (or at least its headers), so that I can try to compare it with the PocketBase one?

**Comment by jscheller** [comment]
Thanks for your quick reply!  Yes, I'm using that 'noreply@wordsearchwizard.com' in the Sender address field in the mail settings in PocketBase, same as fromAddr in that Python script.

Here's a copy of the message source from one of the emails sent via Python from my PocketBase container delivered successfully. I've redacted the delivery address, but everything else is unchanged...

```
Delivered-To: [REDACTED]@gmail.com
Received: by 2002:a05:6a06:1ea3:b0:68f:8310:6360 with SMTP id z35csp29195pis;
        Mon, 20 Nov 2023 12:04:13 -0800 (PST)
X-Google-Smtp-Source: AGHT+IHtS+rqzhy6KH4sCOpN5fUtOyF//jJ+p8SEJ/wxfVDtEiSOuwE0/fhQKXB6RK3IMtnTapeU
X-Received: by 2002:a05:6000:178f:b0:332:ca2d:c7aa with SMTP id e15-20020a056000178f00b00332ca2dc7aamr2500126wrg.40.1700510653206;
        Mon, 20 Nov 2023 12:04:13 -0800 (PST)
ARC-Seal: i=1; a=rsa-sha256; t=1700510653; cv=none;
        d=google.com; s=arc-20160816;
        b=F/srYdu9WUg4sgioVrOmsRsg0Hkz3QHlrBWf/OPCIkU9Ulp7UY0U42s/O/p0W+G+9p
         L3YhA0ZupgH4FBVmlda2jzGcIkGzt5/osFMrYnHYyQhuWMNYElwC0oXKw1lt87KYkzbY
         n1wxYfJENiYJx5WorGGXz2gGalYCodSC6lmrkzMBXYSr8piraWbJeQW3Y1tcpXJuqhA+
         90KRBgNOFpq2htzKII+T3atEjzIfKlbBk7dHrBfUS5D9+ufG2EyleuGzI7pC6L3CpiMO
         EYf4ZKIWR+IKbVkZjQsc6zILa/2q0asiEDVMNc1D9x+/lvDhoDvQU2YGhZdXP6XwUncC
         ikoA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=feedback-id:list-unsubscribe:list-unsubscribe-post:date:subject:to
         :from:mime-version:message-id:dkim-signature:dkim-signature;
        bh=Qrhn3TZakNepech8zVEB4FY60xm7EKS7PAeSadK7qvs=;
        fh=baiTniD4BGZfDR4QoXdn89tMYdJE7PgCD8p1zs+oB/A=;
        b=XjXIoDYsYKBFE0/XTnunO5I2SlL6g8Z3UKRypqTmd7nhcXMZPCqhjN8DxJrQPJAGM7
         4HVglvhTTptyPKkO8nIsT2VRzusdAni7mXSJYYK4FffbG9cGSIPJJJOZEtj3sJ4jdzwq
         dZNFHAlrzP7k1SQubzCBpHnjCUsdolMWLkbnboA7WRJZET4qeHDLaPop9LOoXCIGWtIV
         lKyLiQqG2hAt/wLVkh6+jgyZSqyQP9SBsM+zNsVY+ffXiVU2GSgUAabBTrMv1V10o4NT
         0gIUVOTpHCGGPGhjEQU7zWrDVFF1dIkyGLsbV+iTo/q/Z4u6VAx49bvRbLVPtVnbHjdl
         p5WA==
ARC-Authentication-Results: i=1; mx.google.com;
       dkim=pass header.i=@wordsearchwizard.com header.s=mailjet header.b=GI7bRDjV;
       dkim=pass header.i=@bnc3.mailjet.com header.s=mailjet1 header.b=NLTlcYe3;
       spf=pass (google.com: domain of 4083744d.avwaacys4leaaaaaaaaaaacimt8aaykhtnqaaaaaaapwrablw7u8@a677548.bnc3.mailjet.com designates 185.189.236.51 as permitted sender) smtp.mailfrom=4083744d.AVwAACys4LEAAAAAAAAAAACImt8AAYKHTnQAAAAAAApWrABlW7u8@a677548.bnc3.mailjet.com
Return-Path: <4083744d.AVwAACys4LEAAAAAAAAAAACImt8AAYKHTnQAAAAAAApWrABlW7u8@a677548.bnc3.mailjet.com>
Received: from o51.p25.mailjet.com (o51.p25.mailjet.com. [185.189.236.51])
        by mx.google.com with UTF8SMTPS id s10-20020a5d510a000000b00332c4a02cc5si2179172wrt.824.2023.11.20.12.04.12
        for <[REDACTED]@gmail.com>
        (version=TLS1_3 cipher=TLS_AES_128_GCM_SHA256 bits=128/128);
        Mon, 20 Nov 2023 12:04:13 -0800 (PST)
Received-SPF: pass (google.com: domain of 4083744d.avwaacys4leaaaaaaaaaaacimt8aaykhtnqaaaaaaapwrablw7u8@a677548.bnc3.mailjet.com designates 185.189.236.51 as permitted sender) client-ip=185.189.236.51;
Authentication-Results: mx.google.com;
       dkim=pass header.i=@wordsearchwizard.com header.s=mailjet header.b=GI7bRDjV;
       dkim=pass header.i=@bnc3.mailjet.com header.s=mailjet1 header.b=NLTlcYe3;
       spf=pass (google.com: domain of 4083744d.avwaacys4leaaaaaaaaaaacimt8aaykhtnqaaaaaaapwrablw7u8@a677548.bnc3.mailjet.com designates 185.189.236.51 as permitted sender) smtp.mailfrom=4083744d.AVwAACys4LEAAAAAAAAAAACImt8AAYKHTnQAAAAAAApWrABlW7u8@a677548.bnc3.mailjet.com
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; q=dns/txt;
  d=wordsearchwizard.com; i=noreply@wordsearchwizard.com; s=mailjet; x=1700517852;
  h=message-id:mime-version:from:from:to:to:subject:subject:date:date:list-unsubscribe-post:list-unsubscribe:
  feedback-id:x-csa-complaints:x-mj-mid:x-mj-smtpguid:x-report-abuse-to:
  content-type;
  bh=Qrhn3TZakNepech8zVEB4FY60xm7EKS7PAeSadK7qvs=;
  b=GI7bRDjVpDCv5ybmMgAc/FggfiPHC3JJh1v8jWbeV6HNDQuq8jFIRGToN
 RrFIyPau08b9GVyiM8k/V5L54kycmoiyc57ldkZEDgMTieHrhyHAV7lqbIN3
 6pdaJyzeIu7EbBGYAH/QQkbyL7Yky9yJRa3tYf4Pu23aIsDRdJH1zM=
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; q=dns/txt;
  d=bnc3.mailjet.com; s=mailjet1; x=1700517852;
  h=message-id:mime-version:from:from:to:to:subject:subject:date:date:list-unsubscribe-post:list-unsubscribe:
  feedback-id:x-csa-complaints:x-mj-mid:x-mj-smtpguid:x-report-abuse-to:
  content-type;
  bh=Qrhn3TZakNepech8zVEB4FY60xm7EKS7PAeSadK7qvs=;
  b=NLTlcYe3pQxmC1JMxYt7cQ9GpAYulvd5Kp2TXhz3zjvhKEFNj4MQpZP1v
 W5DvUUyEH6/tctL6IvsIB26WLN+OhX0O2y6Bu+6cVSoeT/s6RM3e/JjOAOtw
 F7XKcbEV9vLyW8HPhkPuHZRNereMMx2diTkCMdiUEsQZyPd5LwTkBI=
Return-Path:
	<4083744d.AVwAACys4LEAAAAAAAAAAACImt8AAYKHTnQAAAAAAApWrABlW7u8@a677548.bnc3.mailjet.com>
Message-Id: <4083744d.AVwAACys4LEAAAAAAAAAAACImt8AAYKHTnQAAAAAAApWrABlW7u8@mailjet.com>
MIME-Version: 1.0
From: noreply@wordsearchwizard.com
To: [REDACTED]@gmail.com
Subject: another simple email in python
Date: Mon, 20 Nov 2023 20:04:12 +0000 (UTC)
List-Unsubscribe-Post: List-Unsubscribe=One-Click
List-Unsubscribe:
	<mailto:unsub-4083744d.r0qy.lt7vpvg3vtwl@bnc3.mailjet.com>,
	<https://redir.dadsworksheets.com/unsub2?m=AVwAACys4LEAAAAAAAAAAACImt8AAYKHTnQAAAAAAApWrABlW7u8A5dSpzAYSKy5TdXTeBHoVQAKASU&b=4083744d&e=5cee0379&x=Rd-a6bH_6_6nFqKu00MsHZFrHE_ZE3D4Hyc-pyea9EY>
Feedback-Id: 42.677548.655653:MJ
X-CSA-Complaints: csa-complaints@eco.de
X-MJ-Mid:
	AVwAACys4LEAAAAAAAAAAACImt8AAYKHTnQAAAAAAApWrABlW7u8A5dSpzAYSKy5TdXTeBHoVQAKASU
X-MJ-SMTPGUID: 039752a7-3018-48ac-b94d-d5d37811e855
X-REPORT-ABUSE-TO: Message sent by Mailjet please report to
	abuse@mailjet.com with a copy of the message
Content-Type: multipart/mixed;
	boundary="===============8766068838729257926=="

--===============8766068838729257926==
Content-Type: text/plain; charset=us-ascii
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit

here is the email
--===============8766068838729257926==--
```

**Reply by ganigeorgiev** [reply]
Thanks for the follow up. Ignoring the gmail headers, the mail looks almost identical to how we construct it with PocketBase. The main difference is that we send both html + text version (the text version is auto generated).

<details>
<summary>Expand to see sample raw PocketBase test reset password mail.</summary>

```
From: Support <support@example.com>
MIME-Version: 1.0
Date: Wed, 15 Nov 2023 16:03:36 +0200
Subject: Reset your Presentator password
To: test@example.com
Message-ID: <eKC7TP15QSdnYsq@example.com>
Content-Type: multipart/mixed;
	boundary="2a7bc449f827a6002bc3a460429c8edfd40a11068619c7bd6f7bf2217028"; charset=UTF-8

--2a7bc449f827a6002bc3a460429c8edfd40a11068619c7bd6f7bf2217028
Content-Type: multipart/alternative;
	boundary="ac6163747ed4423e83eca6913a936a63c4936f49de172c45cdb9e2ce0fb7"

--ac6163747ed4423e83eca6913a936a63c4936f49de172c45cdb9e2ce0fb7
Content-Transfer-Encoding: quoted-printable
Content-Type: text/plain; charset=UTF-8

Hello,=20
Click on the button below to reset your password.=20
[Reset password](http://localhost:5173/_/#/confirm-password-reset/eyJhbGciO=
iJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb2xsZWN0aW9uSWQiOiJfcGJfdXNlcnNfYXV0aF8iLCJ=
lbWFpbCI6InRlc3RAZXhhbXBsZS5jb20iLCJleHAiOjE3MDAwNTg4MTYsImlkIjoiemhmeThuNm=
g0cnJsb2FyIiwidHlwZSI6ImF1dGhSZWNvcmQifQ.yj2qV2KqvoryZe5nIap4ElWFuNxMOslmuG=
0hARGsCPw) =20
If you didn't ask to reset your password, you can ignore this email.=20
Thanks,
Presentator team
--ac6163747ed4423e83eca6913a936a63c4936f49de172c45cdb9e2ce0fb7
Content-Transfer-Encoding: quoted-printable
Content-Type: text/html; charset=UTF-8


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org=
/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns=3D"http://www.w3.org/1999/xhtml">
<head>
    <meta http-equiv=3D"Content-Type" content=3D"text/html; charset=3DUTF-8=
" />
    <meta name=3D"viewport" content=3D"width=3Ddevice-width,initial-scale=
=3D1" />
    <style>
        body, html {
            padding: 0;
            margin: 0;
            border: 0;
            color: #16161a;
            background: #fff;
            font-size: 14px;
            line-height: 20px;
            font-weight: normal;
            font-family: Source Sans Pro, sans-serif, emoji;
        }
        body {
            padding: 20px 30px;
        }
        strong {
            font-weight: bold;
        }
        em, i {
            font-style: italic;
        }
        p {
            display: block;
            margin: 10px 0;
            font-family: inherit;
        }
        small {
            font-size: 12px;
            line-height: 16px;
        }
        hr {
            display: block;
            height: 1px;
            border: 0;
            width: 100%;
            background: #e1e6ea;
            margin: 10px 0;
        }
        a {
            color: inherit;
        }
        .hidden {
            display: none !important;
        }
        .btn {
            display: inline-block;
            vertical-align: top;
            border: 0;
            cursor: pointer;
            color: #fff !important;
            background: #16161a !important;
            text-decoration: none !important;
            line-height: 40px;
            width: auto;
            min-width: 150px;
            text-align: center;
            padding: 0 20px;
            margin: 5px 0;
            font-family: Source Sans Pro, sans-serif, emoji;;
            font-size: 14px;
            font-weight: bold;
            border-radius: 6px;
            box-sizing: border-box;
        }
    </style>
</head>
<body>
    <p>Hello,</p>
<p>Click on the button below to reset your password.</p>
<p>
  <a class=3D"btn" href=3D"http://localhost:5173/_/#/confirm-password-reset=
/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb2xsZWN0aW9uSWQiOiJfcGJfdXNlcnNfY=
XV0aF8iLCJlbWFpbCI6InRlc3RAZXhhbXBsZS5jb20iLCJleHAiOjE3MDAwNTg4MTYsImlkIjoi=
emhmeThuNmg0cnJsb2FyIiwidHlwZSI6ImF1dGhSZWNvcmQifQ.yj2qV2KqvoryZe5nIap4ElWF=
uNxMOslmuG0hARGsCPw" target=3D"_blank" rel=3D"noopener">Reset password</a>
</p>
<p><i>If you didn't ask to reset your password, you can ignore this email.<=
/i></p>
<p>
  Thanks,<br/>
  Presentator team
</p>
</body>
</html>

--ac6163747ed4423e83eca6913a936a63c4936f49de172c45cdb9e2ce0fb7--

--2a7bc449f827a6002bc3a460429c8edfd40a11068619c7bd6f7bf2217028--
```
</details>

My guess is that for some reason the PocketBase mail is being flagged (maybe because of the token link in the html version or the html format, but when I tested it with the default spamassassin filters the spam score was 0).

Just for the test, I've created a trial Mailjet account. The only Mailjet settings that I've changed were:
- Generated SMTP key
- Changed the sender address type to "Website emails/Transactional"

After that I configured PocketBase to use the Mailjet SMTP credentials and sender and tried to send 2 test emails:
![pb_mail_settings](https://i.imgur.com/O3t7qpC.png)
_(note the PLAIN auth; LOGIN auth I think is used only by Microsoft)_

My first test email was to the same email address as the sender and was delivered almost immediately.
The second test email was to my gmail account and in the Mailjet dashboard was first marked as "Queued" and couple minutes later as "Soft-bounced":
![mailjet_sends](https://i.imgur.com/DBTxKEN.png)

I believe the gmail bounce is because I was sending from localhost and didn't bother updating or validating my sender domain (pocketbase.io) DKIM, SPF, DMARC DNS records.

But the important thing is that in both cases the sent test emails are showing in the Mailjet web UI.
So double check your PocketBase configurations and make sure that there are no typo/spaces in the sender address.
Other then that I'm out of ideas why it works in your case with the python script and not with PocketBase especially if they are sent from within the same container and configurations.

_Edit: I forgot to mention that the above tests were done using PocketBase v0.19.4 but there are no changes to the SMTP integration compared to the v0.18.9 release._

**Reply by jscheller** [reply, marked as answer]
Played with this a bunch more this morning, including rebuilding the docker container a few times (including with a completely fresh pb_data directory) and upgrading to v0.19.4, triple checking I had everything setup without hidden space chars, and still couldn't ever get a email out to MailJet. That python script inside the container with what I think is the same config will send, MailJet shows as queued then delivered, but never got a sign MailJet would even see messages sent from the PocketBase admin...  Totally baffling.

Honestly though, Docker seems like it may be just a bit of overkill for a local dev environment for PocketBase, so I just downloaded the MacOS build and ran it directly from a shell in the same directory where my container lives. Went through the mail settings setup again and... it worked just fine. 

Still no real clue why it misbehaved in the container, but it's working and my production deployment wouldn't be running Docker in any case so no concerns here.

Appreciate the work on this project... This is a side project for me that I'm migrating from an ancient Django backend, first pass tried AppWrite (too heavy/complicated for my needs), and on a whim tried out Pocketbase, and wow, apart from this one minor hiccup, PocketBase has been a huge win for this... Thank you!
